src/traffic_cop/traffic_cop.cpp

```diff
@@ -160,28 +162,17 @@ executor::ExecuteResult TrafficCop::ExecuteHelper(
     return p_status_;
   }
 
-  struct ExecutePlanArg {
-    std::shared_ptr<planner::AbstractPlan> plan_;
-    concurrency::TransactionContext *txn_;
-    const std::vector<type::Value> &params_;
-    std::vector<ResultValue> &result_;
-    const std::vector<int> &result_format_;
-    executor::ExecuteResult &p_status_;
-  };
-  auto exec_plan_arg = new ExecutePlanArg{.plan_ = std::move(plan),
-                                          .txn_ = txn,
-                                          .params_ = params,
-                                          .result_ = result,
-                                          .result_format_ = result_format,
-                                          .p_status_ = p_status_};
-
-  threadpool::MonoQueuePool::GetInstance().SubmitTask([](void *arg_ptr) {
-    auto arg = reinterpret_cast<ExecutePlanArg *>(arg_ptr);
-    executor::PlanExecutor::ExecutePlan(arg->plan_, arg->txn_, arg->params_,
-                                        arg->result_, arg->result_format_,
-                                        arg->p_status_);
-    delete (arg);
-  }, exec_plan_arg, task_callback_, task_callback_arg_);
+  auto &pool = threadpool::MonoQueuePool::GetInstance();
+  pool.SubmitTask([plan, txn, &params, &result, &result_format, this] {
+    executor::PlanExecutor::ExecutePlan(
+        plan, txn, params, result_format,
+        [&result, this](executor::ExecuteResult &&p_status,
+                        std::vector<ResultValue> &&values) {
+          this->p_status_ = p_status;
+          result = std::move(values);
+          task_callback_(task_callback_arg_);
```

Comment:
You're constructing a nested lamba-function here which can be confusing. Can you break these up?